File: AndroidStudioProjects/CamaraXApp/app/src/main/java/com/example/camaraxapp/MainActivity.java
package com.example.camaraxapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.content.Intent;
import android.provider.MediaStore;
import android.graphics.Bitmap;
import android.app.Activity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    //Cosas para tomar una foto

    static final int REQUEST_IMAGE_CAPTURE = 1;

    // Invoca un intent para capturar una foto.
    private void dispatchTakePictureIntent() {
        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
        if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
            startActivityForResult(takePictureIntent, REQUEST_IMAGE_CAPTURE);
        }
    }
    //recupera esta tomada imagen y la toma como un Bitmap pequeño
    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == REQUEST_IMAGE_CAPTURE && resultCode == RESULT_OK) {
            Bundle extras = data.getExtras();
            Bitmap imageBitmap = (Bitmap) extras.get("data");
            //No necesitas mostrar la imagen, solo obtenerla que ya lo haces en imagebitmap
            //imageView.setImageBitmap(imageBitmap);
        }
    }


}

File: AndroidStudioProjects/helloword/app/src/main/java/com/example/helloword/MainActivity.kt
package com.example.helloword

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        PermissionHelper.checkAndRequestCameraPermissions(this);
        setContentView(R.layout.activity_main)
    }
    
}fun onRequestPermissionsResult(
    requestCode: Int, permissions: Array<String?>?, grantResults: IntArray?
) {
    //super.onRequestPermissionsResult(requestCode, permissions, grantResults)
    if (grantResults != null) {
        PermissionHelper.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }
}

fun onResume() {
    //super.onResume()
    if (PermissionHelper.cameraPermissionsGranted(this)) {
        startCamera()
    }
}

fun startCamera() {}